src/tools/rbd_mirror/ImageReplayer.cc

```diff
@@ -1789,30 +1754,53 @@ void ImageReplayer<I>::resync_image(Context *on_finish) {
 
 template <typename I>
 void ImageReplayer<I>::register_admin_socket_hook() {
-  if (m_asok_hook != nullptr) {
-    return;
-  }
+  AdminSocketHook *asok_hook = nullptr;
```

Comment:
Nit: just declare this as ```ImageReplayerAdminSocketHook``` so you don't need to do the static cast below (and you can just assign ```m_asok_hook = asok_hook``` since you know ```m_asok_hook``` is null).